gollm/factory.go

```diff
@@ -193,19 +193,12 @@ func DefaultIsRetryableError(err error) bool {
 // createCustomHTTPClient returns an *http.Client that optionally skips SSL certificate verification.
 // This is shared by all providers that need custom HTTP transport.
 func createCustomHTTPClient(skipVerify bool) *http.Client {
-	if !skipVerify {
-		httpClient := http.DefaultClient
-		httpClient.Timeout = 60 * time.Second
-		return httpClient
-	}
+	transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Not sure if you want to get this http client related changes in with the `list-*` task changes. Looks like this is related to http tracing.

The change skips setting `Proxy: http.ProxyFromEnvironment` which will disable honoring the http proxy environment variables (a bug).